dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -3790,6 +3814,365 @@ def test_null_row_stats_correct_after_updates(self, *mocks):
+        self.assertEqual(259, len(self.trained_schema_full.hashed_row_object))
+        self.assertEqual(500, self.trained_schema_full.total_samples)
+        self.assertEqual(241, self.trained_schema_full._get_duplicate_row_count())
+
+        # not completely accurate since register count is 1024
+        self.assertEqual(259, self.trained_schema_hll.hashed_row_object.cardinality())
+
+    def test_correct_duplicate_row_count_unique_count_disabled(self):
+        profiler_options_1 = ProfilerOptions()
+        profiler_options_1.set(
+            {
+                "*.is_enabled": False,
+            }
+        )
+        profiler = StructuredProfiler(pd.DataFrame([]), options=profiler_options_1)
+        self.assertIsNone(profiler._get_duplicate_row_count())
+
+    def test_correct_duplicate_row_count_empty_profiler(self):
+        profiler_options = ProfilerOptions()
+        profiler_options.set(
+            {
+                "*.is_enabled": False,
+                "row_statistics.is_enabled": True,
+            }
+        )
+        profiler = StructuredProfiler(pd.DataFrame([]), options=profiler_options)
+        self.assertEqual(0, profiler._get_duplicate_row_count())
+
+    def test_duplicate_row_count_cardinality_greater_than_total_samples(self, *mocks):
+        profiler_options = ProfilerOptions()
+        profiler_options.set(
+            {
+                "*.is_enabled": False,
+                "row_statistics.is_enabled": True,
+                "row_statistics.unique_count.is_enabled": True,
+                "row_statistics.unique_count.hashing_method": "hll",
+            }
+        )
+        with mock.patch(
+            "dataprofiler.profilers.profile_builder.HyperLogLog",
+            spec=dataprofiler.profilers.profile_builder.HyperLogLog,
+        ) as hll_mock:
+            hll_mock.return_value.cardinality.return_value = 1000
+            profiler = StructuredProfiler(pd.DataFrame([]), options=profiler_options)
+
+        self.assertEqual(1000, profiler.hashed_row_object.cardinality())
+        self.assertEqual(0, profiler._get_duplicate_row_count())
+
+    def test_save_and_load_hll(self):
+        self.assertEqual(259, self.trained_schema_hll.hashed_row_object.cardinality())
+
+        # Save and Load profile with Mock IO
+        with mock.patch("builtins.open") as m:
+            mock_file = setup_save_mock_open(m)
+            self.trained_schema_hll.save()
+
+            mock_file.seek(0)
+            with mock.patch("dataprofiler.profilers.profile_builder." "DataLabeler"):
+                load_profile = dp.StructuredProfiler.load("mock.pkl")
+        self.assertEqual(259, load_profile.hashed_row_object.cardinality())
+
+        # Check that reports are equivalent
+        save_report = test_utils.clean_report(self.trained_schema_hll.report())
+        load_report = test_utils.clean_report(load_profile.report())
+        self.assertDictEqual(save_report, load_report)
```

Comment:
instead of full report, maybe jsut validate the hll since that is what we care about.

Reply:
done